packages/react/toolbar/src/Toolbar.tsx

```diff
@@ -0,0 +1,246 @@
+import * as React from 'react';
+import { RovingFocusGroup, useRovingFocus } from '@radix-ui/react-roving-focus';
+import { Primitive } from '@radix-ui/react-primitive';
+import { getSelector } from '@radix-ui/utils';
+import { createContext, composeEventHandlers } from '@radix-ui/react-utils';
+import type * as Polymorphic from '@radix-ui/react-polymorphic';
+import type { Merge } from '@radix-ui/utils';
+
+/* -------------------------------------------------------------------------------------------------
+ * Toolbar
+ * -----------------------------------------------------------------------------------------------*/
+
+const TOOLBAR_NAME = 'Toolbar';
+const TOOLBAR_CONTEXT_NAME = TOOLBAR_NAME + 'Context';
+const TOOLBAR_DEFAULT_TAG = 'div';
+
+type Orientation = React.AriaAttributes['aria-orientation'];
+type Direction = 'ltr' | 'rtl';
+
+export type RovingFocusGroupOptions = {
+  orientation?: Orientation;
+  dir?: Direction;
+  loop?: boolean;
+};
+
+type ToolbarOwnProps = Merge<
+  Polymorphic.OwnProps<typeof Primitive>,
+  {
+    /**
+     * The orientation of the toolbar.
+     * Mainly so arrow navigation is done accordingly (left & right vs. up & down)
+     * (default: horizontal)
+     */
+    orientation?: React.AriaAttributes['aria-orientation'];
+    /**
+     * The direction of navigation between toolbar items.
+     * (default: ltr)
+     */
+    dir?: Direction;
+    /**
+     * Whether keyboard navigation should loop focus
+     * (default: true)
+     */
+    loop?: boolean;
+  }
+>;
+
+type ToolbarPrimitive = Polymorphic.ForwardRefComponent<
+  typeof TOOLBAR_DEFAULT_TAG,
+  ToolbarOwnProps
+>;
+
+type ToolbarContextValue = {
+  name: typeof TOOLBAR_CONTEXT_NAME;
+  orientation: Orientation;
+};
+
+const [ToolbarContext, useToolbarContext] = createContext<ToolbarContextValue>(
+  TOOLBAR_CONTEXT_NAME,
+  TOOLBAR_NAME
+);
+
+const Toolbar = React.forwardRef((props, forwardedRef) => {
+  const {
+    children,
+
+    'aria-label': ariaLabel,
+    'aria-labelledby': ariaLabelledBy,
+    'aria-controls': ariaControls,
+
+    orientation = 'horizontal',
+    dir,
+    loop,
+
+    ...ToolbarProps
+  } = props;
+
+  const context: ToolbarContextValue = React.useMemo(() => {
+    return {
+      name: TOOLBAR_CONTEXT_NAME,
+      orientation,
+    };
+  }, [orientation]);
+
+  return (
+    <ToolbarContext.Provider value={context}>
+      <RovingFocusGroup orientation={orientation} dir={dir} loop={loop}>
+        <Primitive
+          ref={forwardedRef}
+          as={TOOLBAR_DEFAULT_TAG}
```

Comment:
It looks like `TOOLBAR_DEFAULT_TAG` might be redundant because `Primitive` is already a `div`.